lib/rack/query_parser.rb

```diff
@@ -177,22 +177,42 @@ def key?(key)
         @params.key?(key)
       end
 
-      def to_params_hash
-        hash = @params
-        hash.keys.each do |key|
```

Comment:
I saved allocating this array of strings for each `Params` object. Since we're not adding keys in the iteration, it's okay to enumerate the hash directly.